### DISCUSSION TITLE
How can I conditionally render components without hydration error?

### DISCUSSION THREAD

**Discussion by odincreek** [vercel/next.js#47801] [Help]
### Summary

PS: Willing to also pay for a little bit mentoring. Telegram: itszhu

I'm using nextjs 13 beta and zustand, whenever I do something like the code to render component based on the zustand store I get a hydration error, how can I make this work without displaying Loading... or something else, just directly what the state is?

This is my client component:
```
import { useAuthStore } from '@/store/store';

const store = useAuthStore((state) => state);
  const user = store.user;

{!user && ( <Component /> )}
{user && ( <OtherComponent /> )}
```

This is my zustand store file:
```

import AsyncStorage from '@react-native-async-storage/async-storage';
import { create } from 'zustand';
import { persist, createJSONStorage } from 'zustand/middleware';

export const useAuthStore = create()(
  persist(
    (set) => ({
      token: '',
      user: null,
      setToken: (token) => set((state) => ({ token: token })),
      setUser: (user) => set(() => ({ ...state, user })),
    }),
    {
      name: 'global',
      storage: createJSONStorage(() => AsyncStorage),
    }
  )
);
```

--------

My second question is how should I approach authentication using access and refresh tokens using express?

Should I store my access token in zustand store and refresh token in httponly cookie?
How would I protect pages and do I need to use middleware, can someone give a simple step-by-step guide on how to approach authentication, currently quiet confused on how to approach this. I would like to let data appear instantly on page load without displaying loading for example and I will also have data that will be updated in realtime.

Can someone push me in the right direction?

**Comment by josegutierro** [comment]
Did you solved?

**Comment by FlyingTurkman** [comment]
Did you tried this?
```
import { useAuthStore } from '@/store/store';

const store = useAuthStore((state) => state);
  const user = store.user;

{!user ? (<Component/>) : (<OtherComponent/>)}
```
